### ISSUE TITLE
Compile error for SAV MK1 (rcbugfix)

### ISSUE THREAD

**Issue by boelle** [MarlinFirmware/Marlin#3935]
when i compile i get this

```
In file included from C:\Users\Bo\AppData\Local\Temp\build8d9b268018e4650011bb6d082bdba173.tmp\sketch\Conditionals.h:264:0,

                 from C:\Users\Bo\AppData\Local\Temp\build8d9b268018e4650011bb6d082bdba173.tmp\sketch\Configuration_adv.h:707,

                 from C:\Users\Bo\AppData\Local\Temp\build8d9b268018e4650011bb6d082bdba173.tmp\sketch\Configuration.h:1178,

                 from C:\Users\Bo\Desktop\Marlin-RCBugFix\Marlin\Marlin.ino:37:

pins.h:143: error: #error "Unknown MOTHERBOARD value set in Configuration.h"
   #error "Unknown MOTHERBOARD value set in Configuration.h"

    ^

In file included from C:\Users\Bo\AppData\Local\Temp\build8d9b268018e4650011bb6d082bdba173.tmp\sketch\Configuration_adv.h:708:0,

                 from C:\Users\Bo\AppData\Local\Temp\build8d9b268018e4650011bb6d082bdba173.tmp\sketch\Configuration.h:1178,

                 from C:\Users\Bo\Desktop\Marlin-RCBugFix\Marlin\Marlin.ino:37:

SanityCheck.h:505: error: #error "HEATER_0_PIN not defined for this board."
   #error "HEATER_0_PIN not defined for this board."

    ^

#error "Unknown MOTHERBOARD value set in Configuration.h"
```

>  #ifndef MOTHERBOARD
>   #define MOTHERBOARD SAV_MKI
> # endif

**Comment by boelle** [comment]
teensduino installed and have selected teensy 2 ++

**Comment by Blue-Marlin** [comment]
How about **BOARD_SAV_MKI**?

**Comment by boelle** [comment]
dooh of course

my only problem left is that the boards.txt is made for an old version of IDE.... and i'm not sure if it matters at all

http://reprap.org/wiki/SAV_MKI

step3...

who have a clue if it makes any difference if i use the custom boards.txt in the zip or i select teensy 2 ++

**Comment by boelle** [comment]
if i just overwrite like the instructions say i get these errors

```
> Board teensy:avr:cdcsav doesn't define a 'build.board' preference. Auto-set to: AVR_CDCSAV
> Board teensy:avr:usbtiny1286 doesn't define a 'build.board' preference. Auto-set to: AVR_USBTINY1286
> Board teensy:avr:avr1091286 doesn't define a 'build.board' preference. Auto-set to: AVR_AVR1091286
> Board teensy:avr:usbtsav doesn't define a 'build.board' preference. Auto-set to: AVR_USBTSAV
> exec: "C:\\Users\\Bo\\Desktop\\arduino-1.6.8\\hardware\\teensy/../tools/": file does not exist
> Error compiling for board [BootloaderCDC]SAV-MkI.
```

**Comment by boelle** [comment]
@thinkyhead are you an "expert" in arduino IDE ?

my board is based on the teensy so as per normal i install teensyduino.. but in ide 1.0.6 i just took a custom boards.txt and smaked in it teensy dir and my board showed up and and the teensy ones vent away...

in ide 1.6.8 that does not work so well and its a bit over my head what i should do

**Comment by thinkyhead** [comment]
> are you an "expert" in arduino IDE ?

Not really! And although I have a Teensy 3.0 I haven't messed with it in a long while. Otherwise I would have more practice installing the Teensyduino support.

I assume you are downloading Version 1.28 and following instructions from: https://www.pjrc.com/teensy/td_download.html ?

**Comment by boelle** [comment]
yep, 

that part is easy enough

hard part is the way i used to install it

that zip file has 2-3 boards.txt examples.... i used the NO PRJ one and put it on desktop... renamed it to boards.txt and used it to overwrite the one that the teensyduino made.. effect was that all the teensy boards did go away and i was left with just my printer electronics

but of course that does not work any longer

have you read the step 3 in the link? 

that will prop say a lot.... but i guess you got my point... the easy fix would just be to go back to 1.0.6 but then again marlin does not support that anymore

I guess the golden words here are that my board is NOT a teensy

but its based on the same chip so uses the teensy lib's

**Comment by boelle** [comment]
hmmmm

i remember that roxy said her printrboard is based on same chip...

i can select teensy 2 ++ like the pins file for printrboard says and it compiles.

but selecting teensy 2 gives option to select mhz and keyboard layout etc.... what to do here?

@Roxy-3DPrintBoard have you tried to upload yet?

**Comment by boelle** [comment]
```
Opening Teensy Loader...

Sketch uses 57.276 bytes (44%) of program storage space. Maximum is 130.048 bytes.
Global variables use 2.922 bytes (35%) of dynamic memory, leaving 5.270 bytes for local variables. Maximum is 8.192 bytes.
Teensy did not respond to a USB-based request to automatically reboot.
Please press the PROGRAM MODE BUTTON on your Teensy to upload your sketch.
```

Got that far.... as per normal i remove the programming jumper and press the reset button...

**Comment by thinkyhead** [comment]
Apparently @paulusjacobus ended up using [Flip](http://www.atmel.com/tools/FLIP.aspx) to get it working on his Printrbot, as detailed a little at https://github.com/MarlinFirmware/Marlin/pull/3182#issuecomment-201260335.

**Comment by boelle** [comment]
will have a look

what is in the way right now i'm just assuming is the teensy loader

when i remove that programming jumper and reset com port changes and it will accept upload directly

**Comment by boelle** [comment]
hmm.. just left a comment there... 

the big difference is the bootloader... and i guess big time that i just have to disable that to get it going... could be wrong thou

**Comment by boelle** [comment]
http://blog.lincomatic.com/?p=548

that one says it all i think.... i have the CDC bootloader

**Comment by boelle** [comment]
https://github.com/scwimbush/Printrboard-HID-Arduino-IDE-Support

might be on to something here

EDIT: my simple head says that maybe that could be used with some renaming  and change of the boards.txt

**Comment by boelle** [comment]
well i tried to upload as is.... 

but it stays on upload... no done msg or timeout

**Comment by boelle** [comment]
dinner.... what could be preventing it from doing the upload right? missing upload protocol??

**Comment by boelle** [comment]
i have a small prayer that @fmalpartida has some spare time to make a boards.txt or something that would make things work in ide 1.6.8

as is @repetier is the only thing i can run

**Comment by tiagom62** [comment]
do you use the usbtinyisp or the bootloader to program the board?

**Comment by paulusjacobus** [comment]
@Boelle I ended up compiling the Marlin files in Arduino IDE with:

 • board: Teensy2++
 • loader: usbtinyisp

The Teensy loader popup appears but I ignored that (never ever worked for me) and took the hex output file from windows app data directory into the Flip programmer. The name of that HEX file will change for each compile you do so select the last one (date).

In FLIP I had to select all 4 check boxes (erase, check, verify ...) in order to flash the hex file to the Rev F Printrbot board. I did set in Flip to board processor to 16MHz (it's the default). This worked for me after 1 day of experimenting, so I know the pain you feel. BTW FLIP is very old software (I believe it is even ported).
